test/test.js

```diff
@@ -1013,4 +1013,35 @@ describe('Overall test', () => {
         done();
     });
 
+    it ('mysql.query should use the same key when retrying', (done) => {
```

Comment:
I think we need both positive & negative test cases?

Reply:
hmm I cant think of a negative test case for this one..